stdlib/_socket.pyi

```diff
@@ -0,0 +1,619 @@
+    def __init__(self, family: int = ..., type: int = ..., proto: int = ..., fileno: Optional[int] = ...) -> None: ...
+    def bind(self, address: Union[_Address, bytes]) -> None: ...
+    def connect(self, address: Union[_Address, bytes]) -> None: ...
+    def connect_ex(self, address: Union[_Address, bytes]) -> int: ...
+    def fileno(self) -> int: ...
+    def getpeername(self) -> _RetAddress: ...
+    def getsockname(self) -> _RetAddress: ...
+    @overload
+    def getsockopt(self, level: int, optname: int) -> int: ...
+    @overload
+    def getsockopt(self, level: int, optname: int, buflen: int) -> bytes: ...
+    if sys.version_info >= (3, 7):
+        def getblocking(self) -> bool: ...
+    def gettimeout(self) -> Optional[float]: ...
+    if sys.platform == "win32":
+        def ioctl(self, control: int, option: Union[int, tuple[int, int, int], bool]) -> None: ...
+    def listen(self, __backlog: int = ...) -> None: ...
+    def recv(self, bufsize: int, flags: int = ...) -> bytes: ...
+    def recvfrom(self, bufsize: int, flags: int = ...) -> tuple[bytes, _RetAddress]: ...
+    if sys.platform != "win32":
+        def recvmsg(self, __bufsize: int, __ancbufsize: int = ..., __flags: int = ...) -> tuple[bytes, list[_CMSG], int, Any]: ...
+        def recvmsg_into(
+            self, __buffers: Iterable[WriteableBuffer], __ancbufsize: int = ..., __flags: int = ...
+        ) -> tuple[int, list[_CMSG], int, Any]: ...
+    def recvfrom_into(self, buffer: WriteableBuffer, nbytes: int = ..., flags: int = ...) -> tuple[int, _RetAddress]: ...
+    def recv_into(self, buffer: WriteableBuffer, nbytes: int = ..., flags: int = ...) -> int: ...
+    def send(self, data: bytes, flags: int = ...) -> int: ...
+    def sendall(self, data: bytes, flags: int = ...) -> None: ...  # return type: None on success
+    @overload
+    def sendto(self, data: bytes, address: _Address) -> int: ...
+    @overload
+    def sendto(self, data: bytes, flags: int, address: _Address) -> int: ...
+    if sys.platform != "win32":
+        def sendmsg(
+            self, __buffers: Iterable[bytes], __ancdata: Iterable[_CMSG] = ..., __flags: int = ..., __address: _Address = ...
+        ) -> int: ...
+    if sys.platform == "linux":
+        def sendmsg_afalg(
+            self, msg: Iterable[bytes] = ..., *, op: int, iv: Any = ..., assoclen: int = ..., flags: int = ...
+        ) -> int: ...
+    def setblocking(self, flag: bool) -> None: ...
+    def settimeout(self, value: Optional[float]) -> None: ...
+    @overload
+    def setsockopt(self, level: int, optname: int, value: Union[int, bytes]) -> None: ...
+    @overload
+    def setsockopt(self, level: int, optname: int, value: None, optlen: int) -> None: ...
+    if sys.platform == "win32":
+        def share(self, process_id: int) -> bytes: ...
+    def shutdown(self, how: int) -> None: ...
+
+SocketType = socket
+
+# ----- Functions -----
+
+if sys.version_info >= (3, 7):
+    def close(fd: int) -> None: ...
```

Comment:
```suggestion
    def close(__fd: int) -> None: ...
```
METH_O

Reply:
Also, this should accept `SupportsIndex`. Probably good to make a type alias for that since this affects all functions here that take an fd argument and pass it to `PyLong_AsSocket_t`.

Reply:
Affects `dup()`, top-level `close()`, and the `socket()` constructor. Changing.